foundry/applications/formApplications/wallConfig.d.ts

```diff
@@ -1,9 +1,90 @@
 /**
  * Wall Configuration Sheet
- *
- * @param object - The Wall object for which settings are being configured
- * @param options - Additional options which configure the rendering of the configuration sheet.
  */
-declare class WallConfig extends FormApplication {
-  _updateObject(event?: Event, formData?: object): Promise<any>; // TODO: Add correct return type
+declare class WallConfig extends FormApplication<WallConfig.Data, Wall> {
+  /**
+   * @param object  - The Wall object for which settings are being configured
+   * @param options - Additional options which configure the rendering of the configuration sheet.
+   */
+  constructor(object: Wall, options?: Partial<WallConfig.Options>);
+
+  options: WallConfig.Options;
+
+  /**
+   * @defaultValue
+   * ```typescript
+   * const options = super.defaultOptions;
+   * options.id = "wall-config";
+   * options.title = "Wall Configuration";
+   * options.template = "templates/scene/wall-config.html";
+   * options.width = 400;
+   * options.editTargets = [];
+   * ```
+   */
+  static get defaultOptions(): WallConfig.Options;
+
+  /**
+   * Provide a dynamically rendered title for the Wall Configuration sheet
+   */
+  get title(): string;
+
+  /**
+   * Construct and return the data object used to render the HTML template for this form application.
+   */
+  getData(): WallConfig.Data;
+
+  /**
+   * This method is called upon form submission after form data is validated
+   * @param event    - The initial triggering submission event
+   * @param formData - The object of validated form data with which to update the object
+   */
+  protected _updateObject(event: Event, formData: WallConfig.FormData): Promise<Wall>;
+}
+
+declare namespace WallConfig {
+  interface Data {
+    object: Duplicated<WallConfig['object']['data']>;
+    options: WallConfig['options'];
+    moveTypes: {
+      [Key in keyof typeof CONST['WALL_MOVEMENT_TYPES'] as typeof CONST['WALL_MOVEMENT_TYPES'][Key]]: Capitalize<
```

Comment:
Might make sense to just create a utility type for this transformation to avoid the repetition. I think it's also used in other places.

Reply:
For posterity: Too complicated to implement.